File: intro/gcd.py
def gcd(m, n):
    if n == 0:
        return m
    return gcd(n, m % n)

def main():
    assert gcd(12, 60) == 12
    assert gcd(0, 343) == 343
    assert gcd(2, 0) == 2
    assert gcd(21, 7) == 7
    assert gcd(17, 17) == 17
    

if __name__ == '__main__':
    main()

File: intro/gcd.c
#include <stdio.h>
#include <assert.h>

int gcd(int m, int n) {
    if (n == 0) return m;
    return gcd(n, m % n);
}

int main() {
    assert(gcd(12, 60) == 12);
    assert(gcd(0, 12) == 12);
    assert(gcd(29, 17) == 1);


    printf("All test passed\n");
    printf("***********************************************\n");
}